ray-on-gke/cloudbuild.yaml

```diff
@@ -64,30 +67,82 @@ steps:
         ray job submit --working-dir ../example_ray_job_scripts \
         --address=http://127.0.0.1:8265 -- python ray_job.py \
         || ( echo "false" >> /workspace/ray_result.txt )
+    waitFor: ['create gke cluster']
 
-        terraform destroy -var=project_id=$PROJECT_ID \
+  - id: 'test jupyterhub'
+    name: 'gcr.io/$PROJECT_ID/terraform'
+    entrypoint: 'bash'
+    args:
+      - '-c'
+      - |
+        cd jupyter-on-gke/tests
+        python3 change_jupyter_config.py
+
+        cd ../
+        terraform apply \
+        -var=project_id=$PROJECT_ID \
         -var=namespace=$SHORT_SHA \
-        -var=service_account=$_USER_NAME-$SHORT_SHA-system-account \
-        -auto-approve -no-color
-    dir: 'ray-on-gke/user/'
-    waitFor: ['get kube config']
+        -var=add_auth=false \
+        -var=enable_iap_service=false \
+        -auto-approve -no-color \
+        || ( echo "false" > /workspace/jupyterhub_tf_result.txt )
+
+        kubectl wait --all pods -n $SHORT_SHA --for=condition=Ready --timeout=300s
+        kubectl get services -n $SHORT_SHA
+        kubectl get service proxy-public -n $SHORT_SHA --output jsonpath='{.status.loadBalancer.ingress[0].ip}' > /workspace/jupyterhub_host_url.txt
+        echo "HOST URL is " $(cat /workspace/jupyterhub_host_url.txt)
+
+        cd tests/
+        python3 test_hub.py $(cat /workspace/jupyterhub_host_url.txt) || ( echo "false" > /workspace/jupyterhub_test_result.txt )
+    waitFor: ['test ray cluster']
 
   - id: 'clean gke cluster'
-    name: 'gcr.io/cloud-builders/gcloud'
+    name: 'gcr.io/$PROJECT_ID/terraform'
     entrypoint: 'bash'
     args:
       - '-c'
       - |
-        gcloud container clusters delete ray-$SHORT_SHA-$_PR_NUMBER-cluster --zone $_ZONE  --project $PROJECT_ID --quiet
+        cd jupyter-on-gke/
+        terraform destroy \
+        -var=project_id=$PROJECT_ID \
+        -var=namespace=$SHORT_SHA \
+        -var=add_auth=false \
+        -var=enable_iap_service=false \
+        -auto-approve -no-color 
+
+        cd ../ray-on-gke/user/
+        terraform destroy -var=project_id=$PROJECT_ID \
+        -var=namespace=$SHORT_SHA \
+        -var=service_account=$_USER_NAME-$SHORT_SHA-system-account \
+        -auto-approve -no-color
+
+        gcloud container clusters delete ray-$SHORT_SHA-$_PR_NUMBER-cluster --zone $_ZONE  --project $PROJECT_ID --quiet 
```

Comment:
why not terraform destry in gke-platform instead of running `gcloud container clusters delete`?

Reply:
Here is I confronted a error for terraform destroy saying cluster's deletion protection is enabled, I tried to find where it came from but failed. So I changed to gcloud command and it went through and I think this should delete everything(?..)
https://screenshot.googleplex.com/8Gh9525zozof4uD

Reply:
I see where the deletion_protection come up. This is relatively new on terraform google provider that it will automatically set `deletion_protection=true` for gke cluster.
According to doc,
`On version 5.0.0+ of the provider, you must explicitly set deletion_protection=false (and run terraform apply to write the field to state) in order to destroy a cluster. It is recommended to not set this field (or set it to true) until you're ready to destroy.`
So to do it in my cloudbuild I need to make `deletion_protection` a variable, and set it to false first and then destroy. but I am hesitate of introducing a new variable to the current template that is not for customers.